builtin/credential/ldap/path_config.go

```diff
@@ -219,6 +258,7 @@ type ConfigEntry struct {
 	BindPassword  string `json:"bindpass" structs:"bindpass" mapstructure:"bindpass"`
 	DiscoverDN    bool   `json:"discoverdn" structs:"discoverdn" mapstructure:"discoverdn"`
 	TLSMinVersion string `json:"tls_min_version" structs:"tls_min_version" mapstructure:"tls_min_version"`
+	Logger        *log.Logger
```

Comment:
Rather than put a Logger here, just make the functions members of `backend` and use its logger.